daml-lf/integration-test-lib/src/main/com/daml/lf/CantonConfig.scala

```diff
@@ -36,19 +37,42 @@ object CantonConfig {
   }
 
   def noTlsConfig = TlsConfiguration(false)
+
 }
 
 final case class CantonConfig(
+    applicationId: ApplicationId,
     darFiles: List[Path] = List.empty,
     authSecret: Option[String] = None,
     devMode: Boolean = false,
     nParticipants: Int = 1,
     timeProviderType: TimeProviderType = TimeProviderType.WallClock,
-    tlsConfig: Option[CantonConfig.Tls] = None,
-    applicationId: ApplicationId = ApplicationId("integrationtest"),
+    tlsEnable: Boolean = false,
     debug: Boolean = false,
 ) {
 
+  lazy val tlsConfig =
+    if (tlsEnable)
+      Some(
+        CantonConfig.Tls(
+          Paths.get(rlocation("test-common/test-certificates/server.crt")),
+          Paths.get(rlocation("test-common/test-certificates/server.pem")),
+          Paths.get(rlocation("test-common/test-certificates/ca.crt")),
+          Paths.get(rlocation("test-common/test-certificates/client.crt")),
+          Paths.get(rlocation("test-common/test-certificates/client.pem")),
+        )
+      )
+    else
+      None
+
+  lazy val participantIds =
+    Iterator
+      .range(0, nParticipants)
+      .map(i => Ref.ParticipantId.assertFromString("participant" + i.toString))
+      .toVector
+
+  lazy val ledgerIds = participantIds.asInstanceOf[Vector[String]]
```

Comment:
Given this abstraction, we may want to consider allowing a user of this to set their own participant id(s).